server/src/main/java/org/elasticsearch/index/mapper/AliasRuntimeField.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Function;
+
+public class AliasRuntimeField implements RuntimeFieldType {
+
+    public static final String CONTENT_TYPE = "alias";
+
+    private static class Builder extends RuntimeFieldType.Builder {
+
+        final FieldMapper.Parameter<String> path = FieldMapper.Parameter.stringParam(
+            "path",
+            true,
+            RuntimeFieldType.initializerNotSupported(),
+            null
+        ).setValidator(
+            s -> {
+                if (s == null) {
+                    throw new MapperParsingException("Missing required parameter [path]");
+                }
+            }
+        );
+
+        protected Builder(String name) {
+            super(name);
+        }
+
+        @Override
+        protected List<FieldMapper.Parameter<?>> getParameters() {
+            return List.of(path);
+        }
+
+        @Override
+        protected RuntimeFieldType buildFieldType() {
+            return new AliasRuntimeField(name, path.get());
+        }
+    }
+
+    public static final RuntimeFieldType.Parser PARSER = new RuntimeFieldType.Parser((n, c) -> new Builder(n));
+
+    private final String name;
+    private final String path;
+
+    public AliasRuntimeField(String name, String path) {
+        this.name = name;
+        this.path = path;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(name);
+        builder.field("type", "alias");
+        builder.field("path", path);
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public MappedFieldType asMappedFieldType(Function<String, MappedFieldType> lookup) {
```

Comment:
interesting approach! out of curiosity (and I don't know what is better, I have not thought this through yet), have you tried/considered implementing alias fields with a MappedFieldType that proxies all of its methods to the corresponding field it points to?

Reply:
That kind of approach is useful when you want to wrap another instance and maybe filter a few things, but in this case the MappedFieldType that we want to return really is just the field type defined by another mapper.  Plus it makes it easy to miss things out when we add another method to the base class.

Re the API leakage, I'm not too concerned, just because all that's leaking really is the fact that runtime fields can refer to other fields.  I think it might be interesting to try and rework the script fields into using this lookup as the basis for their SearchLookup instances as well, which would mean that we could remove the `forkAndTrackFieldReferences()` method.  It also opens up the space for other implementations - for example, we could add an 'adapter' field type which refers to another field and applies a simple adapter script to all its values, removing the need to handle empty or multiple values in runtime scripts.  And we've talked about having text fields with different similarities or boosts on them, which could be implemented like this.